services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -12,15 +14,49 @@ import scala.concurrent.{ ExecutionContext, Future }
 class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
     extends ProjectVersioning[VersioningException] {
 
-  override def updateFile(project: Project, projectFile: ProjectFile)(
+  override def updateFile(project: Project, projectFile: ProjectFile, version: Option[Version])(
     implicit ec: ExecutionContext
-  ): AsyncResult[String] = ???
+  ): AsyncResult[String] = {
+    val path = URLEncoder.encode(projectFile.path.toString, "UTF-8")
+    val fileUrl = s"${config.url}projects/${project.repository}/repository/files/$path"
+    val fileVersion = version match {
+      case Some(version) => version.value
+      case None          => config.defaultFileVersion
```

Comment:
You're doing pattern matching on `version` and inside you get `version`. It's confusing naming.

Reply:
Solved